Src/Support/Google.Apis.Auth.Tests/OAuth2/ImpersonatedCredentialTests.cs

```diff
@@ -0,0 +1,230 @@
+            var credential = new ImpersonatedCredential(initializer);
+            Assert.Equal(3600, credential.LifetimeInSeconds);
+            Assert.Equal(delegates, credential.Delegates);
+            Assert.Equal(scopes, credential.Scopes);
+            Assert.Equal("foo", credential.TargetPrincipal);
+        }
+
+        [Fact]
+        public async Task TestFetchesOidcToken()
+        {
+            var credential = CreateImpersonatedCredentialWithIdTokenResponse();
+            var oidcToken = await credential.GetOidcTokenAsync(OidcTokenOptions.FromTargetAudience("will.be.ignored"));
+            var token = await oidcToken.GetAccessTokenAsync();
+            Assert.Equal(674146, oidcToken.TokenResponse.ExpiresInSeconds);
+            Assert.Equal(OidcComputeSuccessMessageHandler.FirstCallToken, token);
+        }
+
+        [Fact]
+        public async Task TestFetchesOidcTokenFailure()
+        {
+            var credential = CreateImpersonatedCredentialWithErrorResponse();
+            var oidcToken = await credential.GetOidcTokenAsync(OidcTokenOptions.FromTargetAudience("will.be.ignored"));
+            var ex = await Assert.ThrowsAsync<TokenResponseException>(async () =>{var result = await oidcToken.GetAccessTokenAsync();});
+            Assert.Equal(errorResponseContent, ex.Error.Error);
+        }
+
+        [Fact]
+        public async Task TestFetchesAccessToken()
+        {
+            var credential = CreateImpersonatedCredentialWithAccessTokenResponse();
+            var success = await credential.RequestAccessTokenAsync(CancellationToken.None).ConfigureAwait(false);
+            Assert.True(success);
+            Assert.Equal(3600, credential.Token.ExpiresInSeconds);
+            Assert.Equal("access_token", credential.Token.AccessToken);
+        }
+
+        [Fact]
+        public async Task TestFetchesAccessTokenFailure()
+        {
+            var credential = CreateImpersonatedCredentialWithErrorResponse();
+            var ex = await Assert.ThrowsAsync<TokenResponseException>(async () =>{await credential.RequestAccessTokenAsync(CancellationToken.None).ConfigureAwait(false);});
+            Assert.Equal(errorResponseContent, ex.Error.Error);
+        }
+
+        [Fact]
+        public async Task TestSignBlob()
+        {
+            var credential = CreateImpersonatedCredentialWithSignBlobResponse();
+            var bytes = await credential.SignBytes(Encoding.ASCII.GetBytes("toSign")).ConfigureAwait(false);
+            // Signature is "foo" whose byte array is {102, 111, 111}.
+            Assert.Equal(new byte[] {102, 111, 111}, bytes);
+        }
+
+        [Fact]
+        public async Task TestSignBlobFailure()
+        {
+            var credential = CreateImpersonatedCredentialWithErrorResponse();
+            var ex = await Assert.ThrowsAsync<ImpersonatedCredential.SignBlobResponseException>(async () =>{await credential.SignBytes(Encoding.ASCII.GetBytes("toSign")).ConfigureAwait(false);});
```

Comment:
No need for a block-bodied lambda here - and no need even to await it.

```csharp
var ex = await Assert.ThrowsAsync<ImpersonatedCredential.SignBlobResponseException>(
    () => credential.SignBytes(Encoding.ASCII.GetBytes("toSign"));
```

Reply:
done